README.md

```diff
@@ -0,0 +1,701 @@
+│                                   │
+│          FESTIVAL INFO            │
+│                                   │
+│    owner=admins──────────────┐    │
+│    organiser:add             │    │
+│    │                         │    │
+│    │         EVENTS          │    │
+│    │                         │    │
+│    │   owner=organiser───┐   │    │
+│    │   │                 │   │    │
+│    │   │      EVENT      │   │    │
+│    │   │                 │   │    │
+│    │   └─────────────────┘   │    │
+│    │                         │    │
+│    │   owner=organiser───┐   │    │
+│    │   │                 │   │    │
+│    │   │      EVENT      │   │    │
+│    │   │                 │   │    │
+│    │   └─────────────────┘   │    │
+│    │                         │    │
+│    │   owner=organiser───┐   │    │
+│    │   │                 │   │    │
+│    │   │      EVENT      │   │    │
+│    │   │                 │   │    │
+│    │   └─────────────────┘   │    │
+│    │                         │    │
+│    │                         │    │
+│    └─────────────────────────┘    │
+│                                   │
+└───────────────────────────────────┘
+```
+
+In the following diagram we can see how non-hierarchical ownership can be modeled with a photo sharing app. The app displays to the user a collection of photos, this isn't an "owned" collection as no peer has overall authority over it, each user will have their own photo collection. All they need is `read` authority for any published photos.
+
+```
+owner=anna─────────┐   owner=billie───────┐
+me:read            │   me:read            │
+│                  │   │                  │
+│      PHOTO       │   │      PHOTO       │
+│                  │   │                  │
+│                  │   │                  │
+└──────────────────┘   └──────────────────┘
+
+owner=claire───────┐   owner=diana────────┐
+me:read            │   me:read            │
+│                  │   │                  │
+│      PHOTO       │   │      PHOTO       │
+│                  │   │                  │
+│                  │   │                  │
+└──────────────────┘   └──────────────────┘
+```
+
+# System context assumptions
+
+We assume a system where resources can be referenced by unique identifiers (for example, via content hashes) and messages between peers are signed using public key cryptography. Optionally, messages can form append-only logs (by including sequence number and backlinks) so as to provide verifiable causal ordering and stricter capability condition restrictions.
+
+# Definitions
+
+**peer:** a user possessing knowledge of the private key for an ed25519 signing keypair; publishes operations and syncs document state with other peers  
+**owner:** the peer who created a document  
+**issuer:** the peer issuing authority via a capability  
+**receiver:** the peer receiving authority via a capability  
+**authorizer:** the peer authorizing a request or operation against a capability  
```

Comment:
This one confuses me. Is it a "verifier" of an operation? Is it possible for operations to be published which are "not valid".

i guess there are:
- valid rpc/ requests
- valid operations

requests often discuss "authorization".. but if an operation has been pulished (offline, elsewhere) authoring has already happened... So all you can do is verify? I am def bringing assumptions from my scuttlebutt implementation that may be wrong

Reply:
I get caught onthis cos "authorize" is "can author". Authoring of messages in p2p context is not same as acceptance though

Reply:
"authorizer" is not related to the author of a resource, but the peer acting as the authority system at any moment. They look at operations they are sent, check that a capability exists which contains the required authority, and then chooses to accept or reject the operation accordingly.

Reply:
https://dictionary.cambridge.org/dictionary/english/authorize